packages/flutter/lib/src/rendering/object.dart

```diff
@@ -2070,16 +2134,34 @@ abstract class RenderObject extends AbstractNode with DiagnosticableTreeMixin im
   /// to repaint.
   ///
   /// If this getter returns true, the [paintBounds] are applied to this object
-  /// and all descendants. The framework automatically creates an [OffsetLayer]
-  /// and assigns it to the [layer] field. Render objects that declare
-  /// themselves as repaint boundaries must not replace the layer created by
-  /// the framework.
+  /// and all descendants. The framework invokes [RenderObject.updateCompositedLayer]
+  /// to create an [OffsetLayer] and assigns it to the [layer] field.
+  /// Render objects that declare themselves as repaint boundaries must not replace
+  /// the layer created by the framework.
   ///
-  /// Warning: This getter must not change value over the lifetime of this object.
+  /// If the value of this getter changes, [markNeedsCompositingBitsUpdate] must
+  /// be called.
   ///
   /// See [RepaintBoundary] for more information about how repaint boundaries function.
   bool get isRepaintBoundary => false;
 
+  /// T render object composites as a repaint boundary, but only if its child
+  /// does.
+  ///
+  /// For example, a render object which is a repaint boundary only when its
+  /// child needs compositing would declare this as follows:
+  ///
+  /// ```dart
+  ///    @override
+  ///    bool get isRepaintBoundary => child?.needsCompositing ?? false;
+  ///
+  ///    @override
+  ///    bool get compositingDependsOnChild => true;
+  /// ```
+  ///
+  /// See [RepaintBoundary] for more information about how repaint boundaries function.
+  bool get compositingDependsOnChild => false;
```

Comment:
This is the new flag necessary to do something like the code snippet in the description. When we mark the child compositing bits dirty, we need to walk up the tree until we hit a repaint boundary. If the parent repaint boundary depends on the child compositing, then we can't ask the parent if its a repaint boundary until the child compositing is finished. To work around this, we always mark the parent compositing bits dirty if it claims to depend on the child compositing state.